internal/flagset/flagset.go

```diff
@@ -133,43 +111,46 @@ func (fs *Flagset) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
-func FormatValidationError(issues []manifest.ValidationError) string {
-	var sb strings.Builder
-	sb.WriteString("flag manifest validation failed:\n\n")
-
-	// Group messages by flag path
-	grouped := make(map[string]struct {
-		flagType string
-		messages []string
-	})
 
-	for _, issue := range issues {
-		entry := grouped[issue.Path]
-		entry.flagType = issue.Type
-		entry.messages = append(entry.messages, issue.Message)
-		grouped[issue.Path] = entry
+func LoadFromSourceFlags(data []byte) (*[]Flag, error) {
+	type SourceFlag struct {
+		Key          string `json:"key"`
+		Type         string `json:"type"`
+		Description  string `json:"description"`
+		DefaultValue any    `json:"defaultValue"`
 	}
 
-	// Sort paths for consistent output
-	paths := make([]string, 0, len(grouped))
-	for path := range grouped {
-		paths = append(paths, path)
+	// First try to unmarshal as an object with a "flags" property
+	var sourceWithWrapper struct {
+		Flags []SourceFlag `json:"flags"`
 	}
-	sort.Strings(paths)
-
-	// Format each row
-	for _, path := range paths {
-		entry := grouped[path]
-		flagType := entry.flagType
-		if flagType == "" {
-			flagType = "missing"
+
+	var sourceFlagsArray []SourceFlag
+
+	if err := json.Unmarshal(data, &sourceWithWrapper); err == nil && len(sourceWithWrapper.Flags) > 0 {
+		// Successfully unmarshaled as object with flags property
+		sourceFlagsArray = sourceWithWrapper.Flags
+	} else {
+		// Try to unmarshal as a direct array of flags (for backward compatibility)
+		if err := json.Unmarshal(data, &sourceFlagsArray); err != nil {
+			return nil, fmt.Errorf("failed to parse flags: expected either {\"flags\": [...]} or direct array format")
 		}
```

Comment:
when the source url directs to the (`sample_manifest.json`)[https://github.com/open-feature/cli/blob/main/sample/sample_manifest.json] the unmarshaling fails. I believe how that manifest is structured with `flags` being a map of the keys not an slice of structs, resulting in a mismatch.
```suggestion
		var manifestFlags Flagset
		if err := json.Unmarshal(data, &manifestFlags); err != nil {
			return nil, fmt.Errorf("failed to parse flags: expected either {\"flags\": [...]} or direct array format")
		}
		return &manifestFlags.Flags, nil
```

Reply:
I saw that, and it could have been me it was later in the night but I juste kept hitting the failed to parse when using that `sample_manifest` until I tried that so I wanted to mention it just it case it wasn’t me 😅😂

Reply:
thank you for pointing this out, it was an issue

I added some additional logic to support both ways